Engine/Rendering/FrameBuffer/FrameBuffer.cpp

```diff
@@ -0,0 +1,126 @@
+#include "FrameBuffer.h"
+
+FrameBuffer::FrameBuffer(int num_color_attachements) : num_color_attachements(num_color_attachements)
+{
+	glGenFramebuffers(1, &fbo);
+}
+
+FrameBuffer::~FrameBuffer()
+{
+	ClearAttachements();
+	glDeleteFramebuffers(1, &fbo);
+}
+
+void FrameBuffer::GenerateAttachements(float width, float height)
+{
+	GenerateColorAttachement(width, height);
+	GenerateDepthAttachement(width, height);
+	LinkAttachements();
+}
+
+void FrameBuffer::ClearAttachements()
+{
+	for (size_t i = 0; i < num_color_attachements; ++i)
+	{
+		if (color_attachements[i] != 0)
+		{
+			glDeleteTextures(1, &color_attachements[i]);
+		}
+	}
+
+	if (rbo != 0)
+	{
+		glDeleteRenderbuffers(1, &rbo);
+	}
+}
+
+void FrameBuffer::Bind(GLenum target) const
+{
+	glBindFramebuffer(target, fbo);
+}
+
+void FrameBuffer::UnBind(GLenum target)
+{
+	glBindFramebuffer(target, 0);
+}
+
```

Comment:
how is that you can bind target to fbo on constant being declarated on the class but you can't put const on the unbid if you are doing the same, binding it to another value?